src/sonic-yang-models/yang-models/sonic-nat.yang

```diff
@@ -0,0 +1,316 @@
+module sonic-nat {
+    namespace "http://github.com/Azure/sonic-nat";
+    prefix snat;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-acl {
+        prefix sacl;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC NAT yang model";
+
+    revision 2021-03-14 {
+        description
+            "Initial revision.";
+    }
+
+    typedef nat-type {
+        type enumeration {
+            enum snat {
+                description "Source NAT.";
+            }
+            enum dnat {
+                description "Destination NAT.";
+            }
+        }
+    }
+
+    typedef ip-address-range {
+        type union {
+          type inet:ipv4-address;
+          type string {
+                          pattern '(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(-(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9]))?';
+                      }
+        }
+
+        description
+          "This type denotes a single IP address or a range of addresses.";
+    }
+
+    container sonic-nat {
+
+        container STATIC_NAPT {
+
+            description "STATIC_NAPT table part of config_db.json";
+
+            list STATIC_NAPT_LIST {
+                key "global_ip ip_protocol global_l4_port";
+
+                leaf global_ip {
+                    description "Global ip for the Static NAPT entry.";
+
```

Comment:
mandatory global_ip, protocol and l4_port

Reply:
Not needed to mention mandatory keyword for global_ip, protocol and l4_port, as by default key parameters are mandatory.